pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go

```diff
@@ -834,58 +819,120 @@ func TestDeploymentWithProxy(t *testing.T) {
 		return &result, nil
 	}
 
-	ds, rcBool, err := manageOpenShiftControllerManagerDeployment_v311_00_to_latest(
-		deployClient,
-		countNodes,
-		workloadcontroller.EnsureAtMostOnePodPerNode,
-		recorder,
-		operatorConfig,
-		"my.co/repo/img:latest",
-		operatorConfig.Status.Generations,
-		proxyLister,
-		specAnnotations,
-	)
-
-	if err != nil {
-		t.Fatalf("unexpected error: %s", err.Error())
-	}
-	if !rcBool {
-		t.Fatal("apply deployment does not think a changes was made")
-	}
-
-	if ds == nil {
-		t.Fatalf("nil deployment returned")
+	tests := []struct {
```

Comment:
IMO there is no need for 2 tests

Reply:
Testing all the cases requires multiple tests and can only be done by mocking the deployment template as it is being loaded from the asset directory. I aligned the test function to use the default setup in one case, a mocked deployment template in the second case. All cases should now be checked.

Reply:
Ack, having multiple tests looks good now, but let's try to streamline that a bit.